### PR TITLE
Swift 3 Support

### PR DESCRIPTION
**Pull request by mikeger**

### CODE REVIEW

File: Source/Public/NSString+MIMEType.swift [hunk 1]
```diff
@@ -23,7 +23,7 @@ import MobileCoreServices;
 public extension NSString {
     
     /// Returns true if the string represents a valid MIME type and conforms to given uniform type identifier (see UTCoreTypes)
-    func zm_conformsToUTI(UTI: CFString) -> Bool {
+    func zm_conformsToUTI(_ UTI: CFString) -> Bool {
```
Comment (Einh06) [inline Source/Public/NSString+MIMEType.swift]: To conform to the Swift3 naming guideline, we could rename this to: `zm_conformsTo(UTI: CFString) -> Bool`

File: Source/Public/NSUUID+Type1.swift [hunk 2]
```diff
@@ -74,23 +73,45 @@ public extension NSUUID {
...
         let nanoseconds100ToSeconds = Double(10000000)
         let unixTimestamp = Double(nanoseconds100SinceUnixTimestamp) / nanoseconds100ToSeconds
-        return NSDate(timeIntervalSince1970: unixTimestamp)
+        return Date(timeIntervalSince1970: unixTimestamp)
     }
 
     /// Returns the comparison result for this NSUUID of type 1 and another NSUUID of type 1
     /// - Requires: will assert if any UUID is not of type 1
-    public func compareWithType1(uuid: NSUUID) -> NSComparisonResult {
+    public func compareWithType1(_ uuid: NSUUID) -> ComparisonResult {
```
Comment (Einh06) [inline Source/Public/NSUUID+Type1.swift]: we could rename this to follow Swift3 Guideline to something like : `compare(type1UUID: UUID) -> ComparisonResult` or `compareWith(type1UUID: UUID) -> ComparisonResult`

File: Source/Public/Sets.swift [hunk 3]
```diff
@@ -63,35 +63,35 @@ extension Set {
...
-        return self.containsObject(obj)
+    public func contains(_ obj: NSObject) -> Bool {
+        return self.contains(obj)
     }
 
 }
 
 // MARK: OrderedSet
-public final class OrderedSet<T  where T : NSObject, T : Hashable> : Equatable, SequenceType {
+public final class OrderedSet<T> : Equatable, Sequence  where T : NSObject, T : Hashable {
```
Comment (Einh06) [inline Source/Public/Sets.swift]: Do we still need that type? `OrderedSet` are available in swift3.

**Comment by mikeger** [comment]
@Einh06 updated for comments

File: Source/Public/Sets.swift (hunk 3 above)
Comment (marcoconti83) [inline Source/Public/Sets.swift]: Is `OrderedSet` a proper struct, or is it just dropping the `NS` in `NSOrderedSet`?

Comment (daehn) [inline Source/Public/Sets.swift]: Should be just dropping the `NS` prefix

File: Source/Public/Sets.swift [hunk 4]
```diff
@@ -63,108 +63,27 @@ extension Set {
...
-    
-    public init(orderedSet: NSOrderedSet) {
-        self.innerSet = orderedSet.copy() as! NSOrderedSet
-    }
-    
-    public init(set: OrderedSet<T>) {
-        self.innerSet = set.innerSet.copy() as! NSOrderedSet
-    }
-    
-    public func minus(set: OrderedSet<T>) -> OrderedSet<T> {
```
Comment (marcoconti83) [inline Source/Public/Sets.swift]: This could be implemented as extension, if it's not there already on the new OrderedSet

**Pull request merged**